ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -94,6 +97,33 @@ export class AnsClient {
     return this.getAddressByDomainName(domain);
   }
 
+  /**
+   * Mint a new Aptos name
+   *
+   * @param account AptosAccount where collection will be created
+   * @param domainName  Aptos domain name to mint
+   * @param years year duration of the domain name
+   * @returns The hash of the transaction submitted to the API
+   */
+  async mintAptosName(account: AptosAccount, domainName: string, years: AnyNumber = 1): Promise<string> {
+    // check if the name is valid
+    if (domainName.match(nameComponentPattern) === null) {
+      throw new Error(`Name ${domainName} is not valid`);
```

Comment:
shouldn't we have something like an ApiError? Libraries shouldn't throw the generic error, that's an error of last resort, afaiu